src/makeInfinityAdder.js

```diff
@@ -1,10 +1,25 @@
 'use strict';
 
-/***
+/*
  * @return {function}
  */
+
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  return function adder(n) {
+    const totalSum = sum;
+
+    if (n === undefined) {
+      sum = 0;
+
+      return totalSum;
+    }
+
+    sum += n;
+
+    return adder;
```

Comment:
Returning `adder` allows for chaining calls, which is a clever design choice. However, ensure that users are aware that calling the function without arguments will reset the sum.